gtsfm/bundle/bundle_adjustment.py

```diff
@@ -130,6 +133,25 @@ def __construct_factor_graph(
 
         # get all the valid camera indices, which need to be added to the graph.
         valid_camera_indices: List[int] = initial_data.get_valid_camera_indices()
+        for (i1, i2), i2Ti1_prior in relative_pose_priors.items():
```

Comment:
This is becoming a very long function. Factor graphs contain shared pointers and hence it is easy to say:
```
graph.push_back(self._reprojection_factors(...))
graph.push_back(self._pose_priors(...))
graph.push_back(self._between_factors(...))
```

Reply:
PS, saw your commit. Stylewise, maybe avoid `get`, `construct`, `construction` (?).

`graph.push_back(self._pose_priors(...))`

is more readable than

`graph.push_back(self._construct_pose_priors(...))`

PS, why do y'all have *double* `_` in front of private methods? I thought it was just one `_`

Reply:
Thanks. Will do.

`__` is used to model private functions, which should not be overriden. `_` is meant for functions which are designed to be overriden in the children classes.